host.go

```diff
@@ -514,46 +473,55 @@ func (h *Host) Provision() error {
 
 	// install docker - until cloudinit we use ubuntu everywhere so we
 	// just install it using the docker repos
-	cmd, err := h.GetSSHCommand("if [ ! -e /usr/bin/docker ]; then curl -sSL https://get.docker.com | sh -; fi")
+	_, stderr, err := h.GetSSHCommand("if [ ! -e /usr/bin/docker ]; then curl -sSL https://get.docker.com | sh -; fi")
 	if err != nil {
-		return err
-	}
-
-	// HACK: the script above will output debug to stderr; we save it and
-	// then check if the command returned an error; if so, we show the debug
+		buf := bytes.NewBuffer([]byte{})
+		buf.ReadFrom(stderr)
 
-	var buf bytes.Buffer
-	cmd.Stderr = &buf
-
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("error installing docker: %s\n%s\n", err, string(buf.Bytes()))
+		return fmt.Errorf("error installing docker: %s\n%s\n", err, buf.String())
 	}
 
 	return nil
 }
 
-func (h *Host) GetSSHCommand(args ...string) (*exec.Cmd, error) {
+func (h *Host) GetSSHCommand(args ...string) (io.Reader, io.Reader, error) {
```

Comment:
this probably should be named `RunSSHCommand` now as it doesn't get you a cmd anymore + it does the execution.